operator/controllers/service_controller.go

```diff
@@ -47,16 +59,216 @@ type ServiceReconciler struct {
-		For(&lifecyclev1alpha1.Service{}).
+		For(&v1alpha1.Service{}).
 		Complete(r)
 }
+
+// func (r *ServiceReconciler) createDeployment(ctx context.Context, service *v1alpha1.Service) (*appsv1.Deployment, error) {
+// 	deployment := &appsv1.Deployment{
+// 		ObjectMeta: metav1.ObjectMeta{
+// 			Annotations: map[string]string{
+// 				"keptn.sh/application": service.Spec.ApplicationName,
+// 				"keptn.sh/service":     service.Name,
+// 			},
+// 			Name:      service.Name + "-" + r.generateSuffix(),
+// 			Namespace: service.Namespace,
+// 			Labels:    service.Spec.DeploymentSpec.Selector.MatchLabels,
+// 		},
+// 		Spec: service.Spec.DeploymentSpec,
+// 	}
+// 	for i := 0; i < 5; i++ {
+// 		if err := r.Create(ctx, deployment); err != nil {
+// 			if errors.IsAlreadyExists(err) {
+// 				deployment.Name = service.Name + "-" + r.generateSuffix()
+// 				continue
+// 			}
+// 			return nil, err
+// 		}
+// 		break
+// 	}
+// 	return deployment, nil
+// }
+
+func (r *ServiceReconciler) generateSuffix() string {
+	uid := uuid.New().String()
+	return uid[:10]
+}
+
+func (r *ServiceReconciler) startPreDeploymentChecks(ctx context.Context, service *v1alpha1.Service) (string, error) {
+	event := &v1alpha1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				"keptn.sh/application": service.Spec.ApplicationName,
+				"keptn.sh/service":     service.Name,
+			},
+			Name:      service.Name + "-" + r.generateSuffix(),
+			Namespace: service.Namespace,
+		},
+		Spec: v1alpha1.EventSpec{
+			Service:     service.Name,
+			Application: service.Spec.ApplicationName,
+			JobSpec:     service.Spec.PreDeplymentChecks.JobSpec,
+		},
+	}
+	for i := 0; i < 5; i++ {
+		if err := r.Create(ctx, event); err != nil {
+			if errors.IsAlreadyExists(err) {
+				event.Name = service.Name + "-" + r.generateSuffix()
+				continue
+			}
+			return "", err
+		}
+		break
+	}
+	return event.Name, nil
+}
+
+func (r *ServiceReconciler) generateK8sEvent(service *v1alpha1.Service, eventType string) *corev1.Event {
+	return &corev1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName:    service.Name + "-" + eventType + "-",
+			Namespace:       service.Namespace,
+			ResourceVersion: "v1alpha1",
+			Labels: map[string]string{
+				"keptn.sh/application": service.Spec.ApplicationName,
+				"keptn.sh/service":     service.Name,
+			},
+		},
+		InvolvedObject: corev1.ObjectReference{
+			Kind:      service.Kind,
+			Namespace: service.Namespace,
+			Name:      service.Name,
+		},
+		Reason:  string(service.Status.Phase),
+		Message: "pre-deployment checks are " + eventType,
+		Source: corev1.EventSource{
+			Component: service.Kind,
+		},
+		Type: "Normal",
+		EventTime: metav1.MicroTime{
+			Time: time.Now().UTC(),
+		},
+		FirstTimestamp: metav1.Time{
+			Time: time.Now().UTC(),
+		},
+		LastTimestamp: metav1.Time{
+			Time: time.Now().UTC(),
+		},
+		Action:              eventType,
+		ReportingController: "service-controller",
```

Comment:
This is not necessary when the service controller is the owner of this event object